napari/utils/tree/group.py

```diff
@@ -0,0 +1,55 @@
+from __future__ import annotations
+
+from typing import Generator, Iterable
+
+from ..events import NestableEventedList
+from ..events.containers._nested_list import MaybeNestedIndex
+from .node import Node
+
+
+class Group(NestableEventedList[Node], Node):
```

Comment:
Looking at this a group is a node that can have children - I'm sort of wondering if most of the time we're just going to want to have groups and not directly use nodes, but we can see how it all shakes out as we go! It is nice to have option just to have Nodes, which are much simpler as they don't inherit the `NestableEventedList`!!

Reply:
> Looking at this a group is a node that can have children

yep!  that's the core "composite" idea there.  everything is a node.

> It is nice to have option just to have Nodes, which are much simpler as they don't inherit the NestableEventedList!!

Yeah, `Layer` is a just a `Node`, it can't have children.  but `LayerGroup` (i.e. the future `LayerList`) will be a `Group` and therefore a `NestableEventedList` _and_ it is a `Layer/Node` (that's the big change that will allow `LayerGroups` to have properties like opacity and visibility and stuff.